playgrounds/angular/src/app/examples/radio-group/radio-group-example.component.ts

```diff
@@ -1,32 +1,13 @@
 import { ChangeDetectionStrategy, Component } from '@angular/core';
-import { BehaviorSubject } from 'rxjs';
 
 @Component({
   selector: 'dx-radio-group-example',
   template: `
-    <dx-radio-group
-      *ngIf="value$ | async as value"
-      [value]="value"
-      (valueChange)="setValue($event)"
-    >
-      <dx-radio-button value="0" label="Option 0"></dx-radio-button>
-      <dx-radio-button value="1" label="Option 1"></dx-radio-button>
-      <dx-radio-button value="2" label="Option 2"></dx-radio-button>
-    </dx-radio-group>
-
-    <div>
-      <span>Selected value: </span>
-      <span>{{value$ | async}}</span>
-    </div>
+    <dx-radio-group-simple></dx-radio-group-simple>
+    <dx-radio-group-customization-components></dx-radio-group-customization-components>
+    <dx-radio-group-customization-templates></dx-radio-group-customization-templates>
   `,
   changeDetection: ChangeDetectionStrategy.OnPush,
```

Comment:
Question: Can you explain why do we force OnPush strategies here and is it related to how our user should use our component? (For example: this code is obligate for the radio-group component properly working)

Reply:
**Why we must use ``` changeDetection.onPush ```?***

Angular  ``` changeDetection.onPush ``` has one specific feature.
If the parent component has ``` onPush``` strategy, all child components will live with ``` onPush ``` strategy too.
And it doesn't matter what kind of strategy set in the ```@Component``` decorator
And if the component works correctly in ``` onPush ``` strategy, it'll work correctly in the default strategy, but not vice versa.

In other words, if we develop Angular components with onPush it will be suitable for all clients, but if we will choose the default strategy, clients with ``` onPush ``` may have troubles with our components.
This is the reason why we must have ``` onPush ``` in all our Angular components :)

**Regarding using it at this place**
- I automatically use define ``` onPush ``` everythere.
Also, we need to add it to ``` schematic ``` (generation of the component template to set ``` onPush ``` by default) and eslint rule to make it required.
- We must use ``` onPush ``` in playground because if the component works in ``` onPush ``` it will work in default strategy, but not vice versa (see previous paragraph )

Reply:
If I understand you right, it is not necessary or optional for our customers to use the onPush strategy. It is up to their decision.

Reply:
Yeah, our customers can use the change detection strategy that they like :)

Reply:
Yes, I will add it via separate PR soon